### PR TITLE
Fix/lm2 294 add field to allow for tenancy environment on plugin install

### PR DESCRIPTION
**Pull request by joelbitar**

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -77,11 +79,14 @@ public function getEnvironment($apiKey = false)
...
-                constant("\Divido\MerchantSDK\Environment::$environment")
-            )) {
-                $environment
-                    = constant("\Divido\MerchantSDK\Environment::$environment");
+
+            $constantName = "\Divido\MerchantSDK\Environment::$environment";
+
+            if (
+                defined($constantName) &&
+                !is_null(constant($constantName))
```
Comment (nunocosta-divido) [inline Helper/Data.php:87]: could we use `empty` instead ?

File: etc/adminhtml/system.xml [hunk 2]
```diff
@@ -7,132 +7,136 @@
                 <field id="api_key" translate="label" type="text" sortOrder="2" showInDefault="1" showInWebsite="1" showInStore="0">
                     <label>API-key</label>
                 </field>
-                <field id="secret" translate="label" type="text" sortOrder="3" showInDefault="1" showInWebsite="1" showInStore="0">
+                <field id="environment_url" translate="label" type="text" sortOrder="3" showInDefault="1" showInWebsite="1" showInStore="0">
+                    <label>Environment URL</label>
+                    <comment><![CDATA[URL Including "http://" to environment]]></comment>
```
Comment (nunocosta-divido) [inline etc/adminhtml/system.xml:12]:
```suggestion
                    <comment><![CDATA[URL Including "https://" to environment]]></comment>
```

File: Helper/Data.php (hunk 1 above)
Comment (joelbitar) [reply Helper/Data.php:87]: I'm not sure why it was is_null before, I'm hesitant to change 😟

File: etc/adminhtml/system.xml (hunk 2 above)
Comment (joelbitar) [reply etc/adminhtml/system.xml:12]: Good catch! Yes!

File: Helper/Data.php (hunk 1 above)
Comment (joelbitar) [reply Helper/Data.php]: @nunocosta-divido yeah, I think empty is fine.

File: Helper/Data.php [hunk 3]
```diff
@@ -119,31 +124,41 @@ public function getPlatformEnv()
...
+        // Getting environment depending on how apiKey looks
         $env = $this->getEnvironment($apiKey);
         if ($this->debug()) {
             $this->logger->info('Get SDK'.$env);
         }
 
+        // Get environment URL from config or calculate one from apikey
+        $environmentUrl = $this->getEnvironmentUrl($apiKey);
+        if ($this->debug()) {
+            $this->logger->info('Environment URL' . $environmentUrl);
```
Comment (nunocosta-divido) [inline Helper/Data.php:147]:
```suggestion
            $this->logger->info('Environment URL ' . $environmentUrl);
```

**Review by nunocosta-divido — DISMISSED** [review]

**Review by nunocosta-divido — APPROVED** [review]

File: Helper/Data.php [hunk 4]
```diff
@@ -60,9 +62,9 @@ public function __construct(
        /**
      * Checks the SDK's Environment class for the given environment type
      *
-     * @param string $apiKey The config API key
+     * @param string|bool $apiKey The config API key (will default to get from settings)
      *
-     * @return void
+     * @return bool|string
      */
     public function getEnvironment($apiKey = false)
```
Comment (andivido) [inline Helper/Data.php:69]: We should probably use the logic written into the Merchant SDK (https://github.com/dividohq/merchant-api-pub-sdk-php/commit/70eca8ffe79b2bec1882afa76ee69aa5478994d6) for this now?

Comment (joelbitar) [reply Helper/Data.php:69]: Yes, will change to use that ASAP.

**Pull request merged**